qiskit/aqua/algorithms/vq_algorithm.py

```diff
@@ -80,6 +86,8 @@ def var_form(self) -> Optional[VariationalForm]:
     def var_form(self, var_form: VariationalForm):
         """ Sets variational form """
         self._var_form = var_form
+        if var_form:
```

Comment:
If don't choose a default varform then it is necessary to cover the `None` type, but then we should add `Optional` in the type hint.

Reply:
I always expected a user to later provide a valid var form (or whatever the component was) to override what might already be set. I was not logically allowing them setting it to None since the typehint never included Optional. Is there really a valid case to allow one to remove (by setting None), rather than simply override whats there?

Reply:
None not supported anymore 👍